### ISSUE TITLE
Would be nice:  Have the base name of the directory available as variable.

### ISSUE THREAD

**Issue by cgmonroe** [asbjornh/vscode-module-templates#24]
Doing some templates for various common Drupal plugins, like Blocks, Forms, etc.  These are done via Add from Template in folder view.  Currently all of them ask for the Module name, which is the directory name the template is being added to.  

It would be nice to either have the directory name as a variable/helper (e.g. {{dirName}}) or to have it available for a default value for the question.

I think there are other languages / frameworks that use the current directory in library or namespace definitions.

TIA if this can be done.

**Comment by asbjornh** [comment]
I'm not sure I understand. Which directory would you like the name of?

**Comment by cgmonroe** [comment]
Drupal Use case example might help:

Have an existing module that needs a Block Plugin class added to it.  This module is in the folder view like:
```
modules
   custom
      my_module
         my_module.info.yml
         my_module.module
```
The my_module directory name is the 'machine name' of the module.

- You right click on my_module and use New from Template option.  

- Select the Drupal Block plugin template. 

- This asks for the module name, class name, admin label, and category info.

This template creates a "stub" class in src/Plugin/Block directory using this info.  E.g. module name is used in Namespace definition and as a 'prefix' on some internal id values.

If the "my_module" directory name was available, the module name question would not have to be asked.  This would make setting it a bit easier and prevent typos from causing problems.

Basically the current working directory name for the template execution.  I would think that if this was used when a template was run via the command palette that the defaultPath setting would be used.

**Comment by asbjornh** [comment]
I’m afraid I still don’t get it. 

According to your example, the end result is a folder named “my_module” with some files named “my_module.<extension>” inside of it. This is very easy to achieve using the `folder` option on the template (as shown in the example in the readme).

Please try to keep the Drupal lingo to a minimum. It only serves to confuse me!

**Comment by cgmonroe** [comment]
Let's put it this way... 

Template file
```
  "my-template": {
    "displayName": "My Template",
    "questions": {
      "name": "Please enter the name of the directory you just right clicked on.",
      "className": "Class Name"
    },
    "files": [
      {
        "name": "src/Plugin/Block/{{className}}.php",
        "content": [
          "<?php",
          "",
          "namespace Drupal\\\\{{name}}\\Plugin\\Block;",
          "..... more stuff / more uses of {{name}}"
        ]
      }
    ]
  }
```
If there was a helper function (or whatever) like {{cwd_used_to_create_files}} then the first question in the template does not need to be asked.  You just use {{cwd_used_to_create_files}} instead of {{name}}.

**Comment by asbjornh** [comment]
I understand what feature you want, but I don’t understand why.

In your first example, the value of the `name` variable and the `className` variable would be the same (“my_module”). In other words, a single question would be enough.

To me it seems like you really want to make the folder “my_module” first by hand and then run the template that creates the “my_module”-files in that folder. This is unnecessary, since the folder and the files have the same name and this extension  can create the folder for you instead.

**Comment by cgmonroe** [comment]
The purpose of the template is to add a class of any name to a subdirectory of an **existing component**.  So the class name and the directory name are different.  I.e., I can use this to add a DisplayFee.php class, then run it again to create DisplayFi.php class, and the like.  

All of these classes will live in the same Namespace that uses the containing root directory name.  This is similar to Java's package statements.. e.g. `package org.apache.commons.lang3;`, which are also directory names in Java's source.

**Comment by asbjornh** [comment]
Ok, that makes sense.

I'll need to think about this. I'm not gonna just throw a magic `dirName` global in there, so I'll need to figure out a way of exposing what you need without compromising on the design that I want.

**Comment by asbjornh** [comment]
I've pushed a new update of this extension that makes this possible. You'll need to do some work of your own though!

The new [context](https://github.com/asbjornh/vscode-module-templates#context-object) object contains the path to the thing that was clicked in the file explorer (if that was how the extension was called) under `context.vscode.clickedItem.path`. The context also exposes the selected template so that the default path can be picked from there.

In order to create your proposed `dirName` variable, you can create your own helper (see [here](https://github.com/asbjornh/vscode-module-templates#module-templateshandlebarsconfig)) that could look something like this:

```js
const path = require("path");

module.exports = {
  helpers: {
    dirName: function () {
      return path.basename(
        this.context.vscode.clickedItem.path ||
          this.context.template.defaultPath
      );
    }
  }
};
```

It can then be called in your templates like so:

```hbs
{{dirName}}
```

Note that this extension can also be called by right clicking a _file_ in the file explorer, which would make the above `dirName` helper return the file name instead of a folder name, which might or might not be an issue for you
